src/main/java/rx/internal/schedulers/EventLoopsScheduler.java

```diff
@@ -117,10 +117,7 @@ public Subscription schedule(Action0 action) {
             if (isUnsubscribed()) {
                 return Subscriptions.unsubscribed();
             }
-            ScheduledAction s = poolWorker.scheduleActual(action, 0, null);
-            
-            serial.add(s);
-            s.addParent(serial);
+            ScheduledAction s = poolWorker.scheduleActual(action, 0, null, serial);
```

Comment:
The problem here was that ```serial.add()``` happened after the action was scheduled and executed thus if there was some recursive work, those ```serial.add()``` calls could happen before this initial call and thus the order in the list was reversed. Since the removal is O(n) and we expected the list to be in the same order as the executions finish, any disorder increases the remove time and a concurrent lock attempt run out of its spin time and ended up parking. The performance degradation was this increased park/unpark activity. The ```scheduleAction``` overload just adds the action to the ```serial``` before it schedules it thus making sure any recursive schedule happens after it.